### ISSUE TITLE
publicOnly throws Missing JSDoc on private constructors or methods

### ISSUE THREAD

**Issue by lublak** [gajus/eslint-plugin-jsdoc#713]
## Expected behavior
Should not grumble at private methods.

## Actual behavior
Currently, the publicOnly is completely ignored for methods in a class.
`error  Missing JSDoc comment jsdoc/require-jsdoc`

## ESLint Config

```json
{
    "env": {
        "browser": true,
        "es2021": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module",
        "project": "./tsconfig.json"
    },
    "plugins": [
        "@typescript-eslint",
        "jsdoc"
    ],
    "rules": {
        "jsdoc/require-jsdoc": [
            "error",
            {
                "publicOnly": true,
                "require": {
                    "ArrowFunctionExpression": true,
                    "ClassDeclaration": true,
                    "ClassExpression": true,
                    "FunctionDeclaration": true,
                    "FunctionExpression": true,
                    "MethodDefinition": true
                }
            }
        ]
    }
}
```

## ESLint sample

```ts
export class Test {
    private constructor() {
    }
    private test():void {

    }
}
```

## Environment

- Node version: v14.15.0.
- ESLint version v7.25.0
- `eslint-plugin-jsdoc` version: 32.3.1

**Comment by lublak** [comment]
i also tried something like this:

```json
{
    "env": {
        "browser": true,
        "es2021": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module",
        "project": "./tsconfig.json"
    },
    "plugins": [
        "@typescript-eslint",
        "jsdoc"
    ],
    "rules": {
        "jsdoc/require-jsdoc": [
            "error",
            {
                "publicOnly": true,
                "require": {
                    "ArrowFunctionExpression": true,
                    "ClassDeclaration": true,
                    "ClassExpression": true,
                    "FunctionDeclaration": true,
                    "FunctionExpression": true,
                    "MethodDefinition": true
                },
                "contexts": [
                    "MethodDefinition:not([accessibility=\"private\"])"
                ]
            }
        ]
    }
}
```

**Comment by brettz9** [comment]
Try dropping or setting `MethodDefinition` to `false`

**Comment by lublak** [comment]
@brettz9
Then we have the problem that he no longer checks a single method. Whether public or private, it is no longer reported.
For me, it is important that every method that is public has a documentation.
The public methods should still be reported if there is no documentation.
For this purpose, publicOnly should be available, shouldn't it?

**Comment by brettz9** [comment]
I've fixed the docs just now, as it looks like it needs to be `MethodDefinition:not([accessibility="private"]) > FunctionExpression`.

**Comment by lublak** [comment]
@brettz9 ahhh thank you :) now it works fine. Thank you.
